inputs/select.vue

```diff
@@ -135,57 +161,48 @@
       }
     },
     methods: {
-      update(option) {
-        if (_.isArray(option)) {
-          // set all existing options in data to false
-          const newData = _.mapValues(_.cloneDeep(this.data), () => false);
-
-          // for each of the selected options, set the related key in the data to true
-          _.forEach(option, (o) => newData[o.value] = true);
-
-          this.$store.commit(UPDATE_FORMDATA, { path: this.name, data: newData });
-        } else if (_.isObject(option)) {
-          // single new checked option
-          this.$store.commit(UPDATE_FORMDATA, { path: this.name, data: option.value });
+      // input may be stored as simply a value (scalar) or the entire input object
+      formatOptionForStore (o) {
+        if (_.isObject(o) && !this.args.storeRawData) {
+          return o[this.keys.value]
+        } else {
+          return o
         }
       },
+      // basically, all input takes the object form
+      formatOptionForInput (o) {
+        // start-case labels for scalar options
+        if (_.isString(o) || _.isNumber(o)) {
+          return {
+            [this.keys.value]: o,
+            [this.keys.label]: _.startCase(o)
+          };
+        } else {
+          return o
+        }
+      },
+      handleInput (value) {
+        const data = (this.args.multiple)
+          ? value.map(this.formatOptionForStore)
+          : this.formatOptionForStore(value)
+        this.$store.commit(UPDATE_FORMDATA, { path: this.name, data });
+      },
       fetchListItems() {
+        // todo: look into "loading" prop for UiSelect
         const listName = this.args.list,
           lists = this.$store.state.lists,
-          items = _.get(lists, `${listName}.items`);
+          list = _.get(lists, listName, {});
         let promise;
 
-        if (items) {
-          promise = Promise.resolve(items);
+        // todo: hmm i feel like we could use a light wrapper for this list retrieval business
+        if (list.items && !list.isLoading && !list.error) {
```

Comment:
this is the bugfix for the list fetching race condition I mentioned the other day